crates/matrix-sdk-common/src/ring_buffer.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::{vec_deque::Iter, VecDeque};
+
+/// A simple fixed-size ring buffer implementation.
+pub struct RingBuffer<T, N: const usize> {
```

Comment:
Why is `N` a const-generic parameter instead of a regular runtime parameter passed to `new`?

Reply:
Because why not ^^?

Reply:
I wanted to make something more “static”. It can be annoying thought, if we need to store it somewhere.

Reply:
Because it restricts the type to use cases where the size is known at compile time, and I don't see any advantages ^^